### ISSUE TITLE
WAVES Token Gated Links [Track 2. Web3.0 Development Grant] 

### ISSUE THREAD

**Issue by jennifertrin** [Waves-Association/grants-program#59]
- **Proposer:** Jennifer Tran
- **Contacts:** 

Email: jennifer@mintgate.io
Telegram: @jennifer_mintgate

- **Github username:** jennifertrin

## 👥  Team 

**Are you applying to implement the grant as an individual developer or do you represent a team?**

As a team

**Are you a startup or an ongoing business?**

Startup

**What’s your team or startup name? What is a website?**

MintGate Website: https://www.mintgate.io/

### Team members (in case applying as a team or startup)

**Name of team leader**

Jonathan Dunlap

**Names of team members**

Jennifer Tran
Luis Egea
David Tomuletiu
Simon Weniger

### Team experience 

**Please describe your (team/startup) development experience and technological stack?** 

The MintGate team has worked together to develop token gating technologies on various chains since July 2020 and formally as a full-time team since November 2020. Individually, our team has numerous years of combined experience in web and software development. The current MintGate platform is built on React with TailwindCSS on the frontend and NodeJS and Typescript in the backend.

Jonathan Dunlap has 15+ years of software development experience in various industries. Previously, he was an engineering lead at Adobe and launched several community-centric startups.

Prior to joining MintGate, Jennifer Tran worked for one year as an analyst and junior web developer at a boutique development studio focused on building enterprise travel technology. She led local developer groups on civic tech and web development. 

Luis is a graduate of various blockchain master classes including the Consensys Academy Blockchain Developer Bootcamp and has been closely researching the space since 2017. He is well-versed in Solidity contract development and is familiar with various SDKs and APIs that aid in blockchain development.

Simon is an award-winning designer and previously worked in design for major brands such as McDonalds and Trolli. He is well-versed in frontend development and UI/UX design. 

**Do(es) you(r) (team/startup) have experience in using Waves technologies?**

Yes, we have experimented with the [WAVES APIs](https://docs.waves.tech/en/building-apps/waves-api-and-sdk/) and [wallet signing](https://docs.waves.tech/en/building-apps/waves-api-and-sdk/client-libraries/signer) to support wallet login.

**Please attach your (team/startup) portfolio (projects you’ve been involved in)** 

We have built a platform where you can create your own personally branded whitelabel with NFTs and token gated links. https://app.mintgate.io/

**Team website**

MintGate Website: https://www.mintgate.io/
Currently Application Site: https://app.mintgate.io/

### Team Code Repos

Our platform repo is currently private but we have open sourced our token gated link creation widget and frontend library

- Widget: https://github.com/MintKudos/mintgate-widget
- Frontend library: https://github.com/MintKudos/mintflow

## 🔍 Overview

Please describe your solution idea by the following structure:

**Short description of the solution you suggest**

Gate audio, video, and web content using WAVES tokens

We would like to support WAVES Blockchain tokens for our token gated link feature. 

Creators can input a web link to an unlisted page such as an unlisted Youtube video or private Soundcloud link and the WAVES token and amount that they would like to gate the content with. Then our platform will generate a new (token-gated) link that will require the user to sign and confirm the token amount that they hold in order to access the content.

We have already done a beta pilot with Harrison First to gate exclusive audio links using the SIGN token on WAVES and his Ethereum-based FIRST token. We had more than 20 fans access his exclusive token-gated link. 

**How does it solve the specified problem?**

Token gating is when ownership of an NFT or social token is required to access content. Think of it as a creator or community creates a blockchain token and then sets up a paywall that only accepts that token to access exclusive content.

Token gated links enable a variety of use cases including: 

- **Distributing content securely** — Since NFTs and social tokens are on the blockchain, it is not easy for an owner to transfer or share access to a token. Therefore, creators and communities can ensure that their content is distributed directly only to the paid end consumer. No more of only one viewer paying and sharing their email and password with millions of other non-paying consumers.

- **Monetization of any content** — Token gating allows anyone to become a creator. And a creator to be anyone, not only focused on musicians and artists. Since you can lock anything behind an NFT, new creators emerge. From a developer who sells NFTs to access a code tutorial to a business professional who sells NFTs for a 1:1 consultant session.

- **Transformation of new media types** — Token gating transforms traditional media types. They have already transformed the standard JPEG. For example, instead of just purchasing a JPEG, you can purchase an image NFT with content behind it.

**How you will integrate this project into Waves / Neutrino.**

We will integrate WAVES.Exchange wallet and [token check balance APIs](https://nodes.wavesnodes.com/api-docs/index.html#/addresses/getRegularBalance) through the WAVES Node Service.

**Why your team is interested in creating this project.**

We are passionate about bringing utility to tokens on all chains. We envision a creator economy where creators can have tokens on different chains including on WAVES. Long-term, we want our platform to become completely blockchain agnostic, where fans themselves will have a seamless experience interacting with and switching between tokens and token gated links no matter what chain the token was minted on. 

Initially, we envision onboarding creators who already have tokens other chains and have them understand the possibilities of using WAVES as alternative chain to solve problems that they face with their fans (ex. high gas fees) or to launch tokens with various use cases. Since creators can create token gated links gated by multiple tokens across chains, we want to enable creators on WAVES to collaborate with other creators with tokens on different chains. 

**What is the target audience of the final solution?**

We aim to work with creators and influencers on WAVES, as well as creators who have tokens on other chains and want to launch a token on WAVES for other use cases. 

We have already have a commitment to use the final solution by musician Harrison First who has his own Ethereum-based token FIRST and collaborates with the creators of SIGN (on WAVES). We will use his use case as an example of how creators can have tokens on various blockchains including using a token on another chain and launching a new token on WAVES or collaborating with creators with tokens on WAVES. 

**What is the “secret sauce” of it?**

We enable creators to create token gated links gated by multiple tokens on various chains and therefore, fuel collaborations between creators, no matter what chain they are on.

**What is the duration of the whole project?**

We anticipate the final integration to take at most two months. 

**Resources required to implement the grant task**

We will utilize the [WAVES Signer APIs](https://docs.waves.tech/en/building-apps/waves-api-and-sdk/client-libraries/signer) and [token check balance APIs](https://nodes.wavesnodes.com/api-docs/index.html#/addresses/getRegularBalance) through the WAVES Node Service.

### Ecosystem Fit

**Are there any other projects similar to yours? If so, how is your project different?**

No, there is currently no WAVES ecosystem project like ours. 

## 🔩 Development Roadmap 

This section should break out the development roadmap into a number of milestones. It helps to describe the functionality we should expect, plus how we can check that such functionality exists in the product. Whenever milestones are delivered, we refer to the roadmap to ensure that everything has been delivered as expected.

**Step #1**: Support Token Gating Links

- Integrate WAVES wallet login and signing into MintGate + testing
- Integrate the WAVES API into MintGate + testing
- Marketing WAVES integration

Within this application we are requesting **Step #1** funding with a total cost of $12,800 (requesting 800 WAVES) and following KPIs to deliver:

- 100 token gated links on WAVES
- 1 major announcements on MintGate Medium and Discord for creators to utilize WAVES to token gate links
- Series of two announcements on MintGate social media (Twitter and Instagram).

### Overview

- **Total Estimated Duration:** Two months
- **Total Costs:** 

**Integration of WAVES Blockchain for token gated links**
$165 an hour * 70 development hours = $11,550

Marketing WAVES Integration with our existing community
$100 an hour * 5 design hours for marketing assets = $500 
$75 an hour * 10 hours for marketing and communication on WAVES integration with our existing community: $750

$11,550 + $500 + $750 = $12,800

Total: $12,800

Requesting 800 WAVES

## Additional Information ➕

[Documentation on user flow of creating a token gated link
](https://mintgate.gitbook.io/mintgate-docs/set-up-token-access/create-token-gated-access/create-token-gated-links)

[Example video of token gated Youtube video](https://www.youtube.com/watch?v=Ig1leAaAZn0) (using Ethereum-based Unlock Protocol tokens but can be applied to WAVES tokens as well).

**Comment by BlackTurtle123** [comment]
Hello!
We have questions how it exactly is working.
=> We would like to have infrastructure/application architecture schema's and workflows.
=> can you describe the whole authentication process and what parts are decentralized and what data you store centralized
==> describe the whole process in steps from the moment the user logs in till he get to see the link
=> how to prevent people to keep visit old urls once they sold the nft or the tokens

**Comment by jennifertrin** [comment]
Thank you so much for the response! We will send all of the updated documentation within a week

**Comment by BlackTurtle123** [comment]
> Thank you so much for the response! We will send all of the updated documentation within a week

We are still awaiting answers

**Comment by BlackTurtle123** [comment]
No feedback so we will close the issue
